src/js/elements/mapchip.js

```diff
@@ -38,18 +46,48 @@ export class MapChip extends Observable {
     handleChoroplethFilter(args) {
         let primaryGroup = args.data.metadata.primary_group;
         let groups = getFilterGroups(args.data.metadata.groups, primaryGroup);
+        this.groups = groups;
+
+        $(mapOptionsClass).find(`${filterRowClass}[data-isextra=true]`).remove();
```

Comment:
Could you please put this into a method? Right now it's not easy to understand what this is for. Creating a method with a descriptive name will help